DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/bridging/test_bridging_learning_address.py

```diff
@@ -0,0 +1,143 @@
+import pytest
+import asyncio
+
+from dent_os_testbed.lib.bridge.bridge_fdb import BridgeFdb
+from dent_os_testbed.lib.bridge.bridge_link import BridgeLink
+from dent_os_testbed.lib.ip.ip_link import IpLink
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_get_traffic_stats,
+    tgen_utils_setup_streams,
+    tgen_utils_start_traffic,
+    tgen_utils_stop_protocols,
+    tgen_utils_stop_traffic,
+    tgen_utils_get_loss,
+    tgen_utils_dev_groups_from_config,
+    tgen_utils_traffic_generator_connect
+)
+
+pytestmark = pytest.mark.suite_functional_bridging
+
+
+@pytest.mark.asyncio
+async def test_bridging_learning_address(testbed):
+    """
+    Test Name: test_bridging_learning_address
+    Test Suite: suite_functional_bridging
+    Test Overview: Verify that bridge dynamic entries learning.
+    Test Author: Kostiantyn Stavruk
+    Test Procedure:
+    1.  Init bridge entity br0.
+    2.  Set ports swp1, swp2, swp3, swp4 master br0.
+    3.  Set entities swp1, swp2, swp3, swp4 UP state.
+    4.  Set bridge br0 admin state UP.
+    5.  Set ports swp1, swp2, swp3, swp4 learning ON.
+    6.  Set ports swp1, swp2, swp3, swp4 flood OFF.
+    7.  Send traffic to swp1, swp2, swp3, swp4 with source macs 
+        aa:bb:cc:dd:ee:11 aa:bb:cc:dd:ee:12 
+        aa:bb:cc:dd:ee:13 aa:bb:cc:dd:ee:14 accordingly.
+    8.  Verify that address have been learned.
+    """
+
+    bridge = "br0"
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print.error("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    traffic_duration = 10
+
+    out = await IpLink.add(
+        input_data=[{device_host_name: [{"device": bridge, "type": "bridge"}]}])
+    assert out[0][device_host_name]["rc"] == 0,  f"Verify that bridge created.\n{out}"
+
+    out = await IpLink.set(
+        input_data=[{device_host_name:  [
+            {"device": port, "master": "br0", "operstate": "up"} for port in ports]},
+            {"device": bridge, "operstate": "up"}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that bridge, bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
```

Comment:
This line is too long. Use a variable for the error message:
```python
err_msg = f"Verify that bridge, bridge entities set to 'UP' state and links enslaved to bridge"
assert out[0][device_host_name]["rc"] == 0, err_msg
```
And I think that `out` can be omitted, since either way we have it printed in logs